test/libsolidity/semanticTests/array/invalid_encoding_for_storage_byte_array.sol

```
@@ -0,0 +1,92 @@
+    function assignToLong() public { x = "1234567890123456789012345678901234567"; }
+    function copyToStorage() public { y = x; }
+    function copyFromStorageShort() public { y = "abc"; x = y; }
+    function copyFromStorageLong() public { y = "abc"; x = y; }
+    function arrayPop() public { x.pop(); }
+    function arrayPush() public { x.push("t"); }
+    function arrayPushEmpty() public { x.push(); }
+    function del() public { delete x; }
+}
+// ----
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncode() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncodePacked() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToMemory() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// indexAccess() -> 0x6100000000000000000000000000000000000000000000000000000000000000
+// arrayPushEmpty()
+// arrayPush()
+// x() -> 0x20, 5, 0x6162630074000000000000000000000000000000000000000000000000000000
+// arrayPop()
+// assignToLong()
+// x() -> 0x20, 0x25, 0x3132333435363738393031323334353637383930313233343536373839303132, 0x3334353637000000000000000000000000000000000000000000000000000000
+// assignTo()
+// x() -> 0x20, 3, 0x6465660000000000000000000000000000000000000000000000000000000000
+// copyFromStorageShort()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyFromStorageLong()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToStorage()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// y() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// del()
+// x() -> 0x20, 0x00
+// invalidateXLong()
+// x() -> FAILURE
+// abiEncode() -> FAILURE
+// abiEncodePacked() -> FAILURE
+// copyToMemory() -> FAILURE
+// indexAccess() -> FAILURE
+// arrayPushEmpty()
```

I guess `.push()` and `.push(x)` should both fail for both invalid encodings.